test/crdsvalidation/kongroute_test.go

```diff
@@ -13,16 +13,22 @@ import (
 )
 
 func TestKongRoute(t *testing.T) {
+	obj := &configurationv1alpha1.KongRoute{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "KongRoute",
+			APIVersion: configurationv1alpha1.GroupVersion.String(),
+		},
+		ObjectMeta: commonObjectMeta,
+	}
+
 	t.Run("cp ref", func(t *testing.T) {
-		obj := &configurationv1alpha1.KongRoute{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "KongRoute",
-				APIVersion: configurationv1alpha1.GroupVersion.String(),
-			},
-			ObjectMeta: commonObjectMeta,
-		}
+		NewCRDValidationTestCasesGroupCPRefChange(t, obj, NotSupportedByKIC).Run(t)
+	})
 
-		NewCRDValidationTestCasesGroupCPRefChange(t, obj).Run(t)
+	t.Run("cp ref, type=kic", func(t *testing.T) {
+		// NOTE: empty cp ref is not allowed in this context because a route can be attached to a service
```

Comment:
Should we add a separate explicit test case verifying that if no CP ref nor Service ref is set, it gives a full `Has to set either controlPlaneRef or serviceRef` error?

Reply:
👍 Added in 4e5332b76d95d1797ba1c24f7d488bc3bfcea23c